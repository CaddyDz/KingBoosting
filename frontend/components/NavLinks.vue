// This entire component is in AppBarComponent.vue
<template>
	<v-container fluid>
		<v-row>
			<!-- Necessary to finish 12 -->
			<v-col md="2"></v-col>
			<v-col md="2">
				<v-menu transition="slide-y-transition" min-width="100%" open-on-hover>
					<template v-slot:activator="{on}">
						<a href="/regular-boosting" v-on="on" class="link">{{ $t('Boosting') }}</a>
					</template>
					<v-row align="center" justify="center">
						<v-col md="3">
							<nuxt-link to="/regular-boosting">
								<img src="/img/tft.png" alt />
								<!-- Vuetify Built-in class -->
								<p class="text-center">{{ $t('Regular Boosting') }}</p>
							</nuxt-link>
						</v-col>
						<v-col md="3">
							<nuxt-link to="/regular-boosting">
								<img src="/img/tft.png" alt />
								<!-- Vuetify Built-in class -->
								<p class="text-center">{{ $t('Regular Boosting') }}</p>
							</nuxt-link>
						</v-col>
					</v-row>
				</v-menu>
			</v-col>
			<v-col md="2">
				<a href="#" :title="$t('Coaching')" class="link">{{ $t('Coaching') }}</a>
			</v-col>
			<v-col md="2">
				<a href="#" :title="$t('Lol Accounts')" class="link">{{ $t('Lol Accounts') }}</a>
			</v-col>
			<v-col md="2">
				<nuxt-link to="/boosters" :title="$t('Boosters')">
					<p class="text-center link nuxt-link">{{ $t('Boosters') }}</p>
				</nuxt-link>
			</v-col>
			<v-col md="2">
				<a href="#" :title="$t('More*')" class="link">{{ $t('More*') }}</a>
			</v-col>
		</v-row>
	</v-container>
</template>

<style scoped>
/* Make the background color of the menu content dark */
.theme--dark {
	background: rgb(0, 0, 0);
	background: linear-gradient(
		180deg,
		rgba(0, 0, 0, 0) 0%,
		rgba(18, 4, 17, 1) 15%,
		rgba(18, 4, 17, 1) 15%,
		rgba(18, 4, 17, 1) 87%,
		rgba(18, 4, 17, 0) 87%
	);
}
/* The purple underground shade */
img:hover {
	background-image: url("/img/service_nav_bg.png");
	background-position: center bottom;
}
/* Remove stubborn underline */
a {
	text-decoration: none !important;
}
/* Make the text above the underground purple shade */
p {
	position: relative;
	bottom: 40px;
}
.nuxt-link {
	bottom: 3px;
}
/* Add underline manually to navbar links text */
.link {
	padding-bottom: 6px;
	border-bottom: 3px solid currentColor;
}
.v-menu__content {
	box-shadow: none !important;
}
</style>

<i18n>
{
	"en": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"fr": {
	    "Boosting": "Booster",
		"Regular Boosting": "Boost régulier",
		"Coaching": "Allenamento",
		"Lol Accounts": "Comptes Lol",
		"Boosters": "Boosters",
		"More*": "Plus*"
	},
	"it": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "Di Più*"
	},
	"de": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"at": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"swiss": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"es": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"nl": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"se": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
	"ne": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	},
    "dk": {
		"Boosting": "Boosting",
		"Regular Boosting": "Regular Boosting",
		"Coaching": "Coaching",
		"Lol Accounts": "Lol Accounts",
		"Boosters": "Boosters",
		"More*": "More*"
	}
}
</i18n>
