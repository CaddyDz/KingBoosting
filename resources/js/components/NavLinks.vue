<template>
	<div class="text-center">
		<v-menu
			transition="slide-y-transition"
			bottom
			offset-y
			open-on-hover
			open-delay="200"
			close-delay="200"
			min-width="100%"
			nudge-bottom="28"
			internal-activator
			dark
		>
			<template v-slot:activator="{ on }">
				<a href="/regular-boosting" v-on="on" class="link">Boosting</a>
			</template>
			<v-row no-gutters class="dark">
				<v-col md="3"></v-col>
				<v-col md="3">
					<v-row>
						<a href="/regular-boosting" class="title">
							<img src="/images/tft.png" alt="Regular Boosting" class="hvr-glow" />
							<span>Regular Boosting</span>
						</a>
					</v-row>
				</v-col>
				<v-col md="3">
					<v-row>
						<a href="/tft-boosting" class="title">
							<img src="/images/tft.png" alt="TFT Boosting" class="hvr-glow" />
							<span>TFT Boosting</span>
						</a>
					</v-row>
				</v-col>
				<v-col md="3"></v-col>
			</v-row>
		</v-menu>
		<a href="#" title="Coaching" class="link">Coaching</a>
		<a href="#" title="Lol Accounts" class="link">Lol Accounts</a>
		<a href="#" title="Boosters" class="link">Boosters</a>
		<a href="#" title="More*" class="link">More*</a>
	</div>
</template>

<script>
export default {
	data: () => ({
		boostings: [{ title: "Regular Boosting" }, { title: "TFT Boosting" }],
		accounts: [
			{ title: "Unranked Smurf Accounts" },
			{ title: "Ranked Accounts" }
		]
	})
};
</script>

<style scoped>
img {
	cursor: pointer;
}

img:hover {
	background-image: url("/img/service_nav_bg.png");
	background-position: center bottom;
}

.dark {
	background-color: #000;
}

.col > .row {
	margin-left: 10%;
	margin-right: 10%;
}

.col {
	padding: 25%;
}

.link {
	text-decoration: none;
	color: white;
	padding-bottom: 6px;
	border-bottom: 3px solid currentColor;
	margin-right: 0.8rem;
	text-align: center;
}

.title {
	text-align: center;
	text-decoration: none !important;
}

span {
	position: relative;
	bottom: 40px;
}
</style>
