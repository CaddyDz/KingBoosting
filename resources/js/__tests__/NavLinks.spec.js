import {
    shallowMount
} from '@vue/test-utils';

import NavLinks from '../components/NavLinks.vue';


describe('NavLinks', () => {
    it('renders', () => {
        let wrapper = shallowMount(NavLinks);
        expect(wrapper.text()).toContain('Boosting');
        expect(wrapper).toMatchSnapshot();
    });
});
