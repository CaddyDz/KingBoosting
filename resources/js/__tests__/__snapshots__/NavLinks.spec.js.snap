// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks renders 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
